vector/src/main/java/im/vector/app/features/navigation/DefaultNavigator.kt

```diff
@@ -355,54 +329,22 @@ class DefaultNavigator @Inject constructor(
     }
 
     override fun openCreateDirectRoom(context: Context) {
-        val intent = when (val currentGroupingMethod = appStateHandler.getCurrentRoomGroupingMethod()) {
-            is RoomGroupingMethod.ByLegacyGroup -> {
-                CreateDirectRoomActivity.getIntent(context)
-            }
-            is RoomGroupingMethod.BySpace -> {
-                if (currentGroupingMethod.spaceSummary != null) {
-                    SpacePeopleActivity.newIntent(context, currentGroupingMethod.spaceSummary.roomId)
-                } else {
-                    CreateDirectRoomActivity.getIntent(context)
-                }
-            }
-            else -> null
-        } ?: return
-        context.startActivity(intent)
+        when (val currentSpace = appStateHandler.getCurrentSpace()) {
+            null -> CreateDirectRoomActivity.getIntent(context)
+            else -> SpacePeopleActivity.newIntent(context, currentSpace.roomId)
+        }.start(context)
     }
 
     override fun openInviteUsersToRoom(context: Context, roomId: String) {
-        when (val currentGroupingMethod = appStateHandler.getCurrentRoomGroupingMethod()) {
-            is RoomGroupingMethod.ByLegacyGroup -> {
-                val intent = InviteUsersToRoomActivity.getIntent(context, roomId)
-                context.startActivity(intent)
-            }
-            is RoomGroupingMethod.BySpace -> {
-                if (currentGroupingMethod.spaceSummary != null) {
-                    // let user decides if he does it from space or room
-                    (context as? AppCompatActivity)?.supportFragmentManager?.let { fm ->
-                        InviteRoomSpaceChooserBottomSheet.newInstance(
-                                currentGroupingMethod.spaceSummary.roomId,
-                                roomId,
-                                object : InviteRoomSpaceChooserBottomSheet.InteractionListener {
-                                    override fun inviteToSpace(spaceId: String) {
-                                        val intent = InviteUsersToRoomActivity.getIntent(context, spaceId)
-                                        context.startActivity(intent)
-                                    }
-
-                                    override fun inviteToRoom(roomId: String) {
-                                        val intent = InviteUsersToRoomActivity.getIntent(context, roomId)
-                                        context.startActivity(intent)
-                                    }
-                                }
-                        ).show(fm, InviteRoomSpaceChooserBottomSheet::class.java.name)
+        when (val currentSpace = appStateHandler.getCurrentSpace()) {
+            null -> InviteUsersToRoomActivity.getIntent(context, roomId).start(context)
+            else -> {
```

Comment:
I know we just had a conversation about not overusing smaller functions to the point that it obscures intent but this is something I'd definitely extract

Reply:
i've updated, take a look now